adam_preprocessing/shape_stroke_extraction.py

```diff
@@ -510,6 +544,35 @@ def get_strokes(self):
                     sub_part=None,
                 )
             )
+
+        # Calculate relative sizes using already-calculated absolute sizes:
+        for i in range(len(data)):
+            data[i]['relative_size'] = {'x_bigger_than': [], 'y_bigger_than': []}
+            for j in range(len(data)):
+                if i == j:
+                    continue
+                else:
+                    other_object_name = data[j]['object_name']
+                    for axis in ('x', 'y'):
+                        # Relative size is a discrete ('bigger_than') relation
+                        if data[i]['stroke_graph']['size'][axis] > \
+                                data[j]['stroke_graph']['size'][axis]:
+                            data[i]['relative_size'][axis+'_bigger_than'].append(other_object_name)
+
+                        # Normalize relative distances by the relevant axis
+                        # sizes of both objects
+                        object_axis_size = data[i]['stroke_graph']['size'][axis]
+                        other_object_axis_size = data[j]['stroke_graph']['size'][axis]
+                        data[i]['relative_distance'][other_object_name][axis+'_offset'] /= (object_axis_size * other_object_axis_size)
```

Comment:
f-string again.

Also, because this is an assignment you don't need the parens for grouping. Except for the new walrus operator `:=`, Python assignments aren't expressions in the sense that `x = expr` can't be treated as having a value (not even `None`) so there's no ambiguity in writing `x /= y * z` -- it has to mean `x = x / (y * z)`.

Reply:
Personally, I find it a bit easier to read at a glance when there are parentheses, event though `/=` is an assignment operator with no value. I can change it if you think it is detracting from the readability

Reply:
No problem then.